### PR TITLE
fix(pkg): remove --dont-poll-system-solver-variables

### LINKED ISSUE #9425: Remove `--dont-poll-systems-solve-variables`
It's the same as unsetting variables in the solver environment.

### PR DESCRIPTION
**Pull request by rgrinberg**
As discussed with @gridbugs

### CODE REVIEW

File: test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t [hunk 1]
```diff
@@ -53,7 +54,11 @@ Create a workspace config that defines separate build contexts for macos and lin
   > EOF
 
 Now the os-specific dependencies are included on their respective systems.
+<<<<<<< HEAD
```
Comment (gridbugs) [inline test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t:57]: accidentally left in conflict markers

File: test/blackbox-tests/test-cases/pkg/prevent-user-setting-with-test.t [hunk 2]
```diff
@@ -14,7 +14,11 @@ Add a workspace that attempts to set the "with-test" variable:
   >   (lock dune.lock)))
   > EOF
 
+<<<<<<< HEAD
```
Comment (gridbugs) [inline test/blackbox-tests/test-cases/pkg/prevent-user-setting-with-test.t:17]: conflict marker

File: test/blackbox-tests/test-cases/pkg/unavailable-packages.t [hunk 3]
```diff
@@ -27,8 +27,13 @@ Set up two build contexts: a default one for linux and another for macos.
 Helper shell function to generate a dune-project file and generate lockdir for both contexts:
   $ solve_project() {
   >   cat >dune-project
+<<<<<<< HEAD
```
Comment (gridbugs) [inline test/blackbox-tests/test-cases/pkg/unavailable-packages.t:30]: conflict marker

File: test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t [hunk 4]
```diff
@@ -25,6 +25,7 @@ dependencies are included.
   $ solve foo
   Solution for dune.lock:
   - foo.0.0.1
+  - foo-linux.0.0.1
```
Comment (gridbugs) [inline test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t:28]: this extra dependency looks suspicious. Do you need to unset `os` in the solver env for this test now?

**Review by gridbugs — CHANGES_REQUESTED** [review]

File: test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t (hunk 4 above)
Comment (rgrinberg) [reply test/blackbox-tests/test-cases/pkg/env-conditional-dependencies.t]: Ah yes, this test no longer makes sense. I added a lock dir with `os` unset.

**Review by gridbugs — APPROVED** [review]

**Pull request merged**